### PR TITLE
internal/providers: implement Teleport kubernetes config provider

### PR DESCRIPTION
**Pull request by edganiukov**
Example of user Teleport config:
```
configure:  {
  name:  "prod-cluster"
  configuration:  {
    [type.googleapis.com/foundation.universe.teleport.configuration.Configuration]:  {
      proxy:  {
        user_profile:  "foobar.teleport.sh"
        kube_cluster:  "prod-cluster"
        registry_app:  "prod-registry"
      }
      registry:  "http://registry.prod.example:5000"
    }
  }
}
```

Example of a bot Teleport config:
```
configure:  {
  name:  "prod-cluster"
  configuration:  {
    [type.googleapis.com/foundation.universe.teleport.configuration.Configuration]:  {
      proxy:  {
        proxy_url:  "foobar.teleport.sh:443"
        kube_cluster:  "prod-cluster"
        kube_certs_dir: "/opt/machine-id/kube"
        registry_app:  "prod-registry"
        registry_certs_dirs: "/opt/machine-id/registry"
      }
      registry:  "http://registry.prod.example:5000"
    }
  }
}
```
Fixes NSL-2450
Towards NSL-2535

### CODE REVIEW

**Review by n-g — APPROVED** [review]

File: internal/providers/teleport/teleport.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+		if err != nil {
+			return nil, fnerrors.UsageError("Login with 'tsh login'", "Teleport profile is not found.")
+		}
+
+		cert, err := parseCertificate(ctx, profile.TLSCertPath())
+		if err != nil {
+			return nil, fnerrors.InternalError("failed to load user's certificate")
+		}
+
+		if cert.NotAfter.Before(time.Now().Add(loginMinValidityTTL)) {
```
[the remark this answers is not in the document]
Comment (edganiukov) [reply internal/providers/teleport/teleport.go]: Nice, changed.

File: internal/providers/teleport/teleport.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+			usage := fmt.Sprintf(
+				"Login with 'tsh login --proxy=%s --user=%s %s'",
+				profile.WebProxyAddr, profile.Username, profile.SiteName,
+			)
+			return nil, fnerrors.UsageError(usage, "Teleport credentials have expired.")
+		}
+
+		tpRegistryApp := conf.GetProxy().GetRegistryApp()
+		if err := tshAppsLogin(ctx, profile, tpRegistryApp); err != nil {
+			fnerrors.InvocationError("tsh", "failed to login to app %q", tpRegistryApp)
```
[the remark this answers is not in the document]
Comment (edganiukov) [reply internal/providers/teleport/teleport.go]: done

File: internal/providers/teleport/teleport.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+		return nil
+	}
+
+	return tasks.Return0(ctx, tasks.Action("tsh.apps.login").Arg("app", app), func(ctx context.Context) error {
+		tshBinPath, err := exec.LookPath(tshBin)
+		if err != nil {
+			return fnerrors.InternalError("missing tsh binary")
+		}
+
+		c := exec.Command(tshBinPath, "apps", "login", app, "--ttl", appLoginTTLMins)
```
[the remark this answers is not in the document]
Comment (edganiukov) [reply internal/providers/teleport/teleport.go]: Done

File: internal/providers/teleport/teleport.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+			return errors.New("app certificate expires soon")
+		}
+
+		return nil
+	}); err == nil {
+		return nil
+	}
+
+	return tasks.Return0(ctx, tasks.Action("tsh.apps.login").Arg("app", app), func(ctx context.Context) error {
+		tshBinPath, err := exec.LookPath(tshBin)
```
[the remark this answers is not in the document]
Comment (edganiukov) [reply internal/providers/teleport/teleport.go]: Right, this was copy pasted. Use `tsh` directly.

**Review by hugosantos — APPROVED** [review]

**Pull request merged**